src/appleseed/foundation/math/sampling/qmcsamplingcontext.h

```diff
@@ -256,7 +293,8 @@ inline void QMCSamplingContext<RNG>::split_in_place(
     const size_t        dimension,
     const size_t        sample_count)
 {
-    assert(m_sample_count == 0 || m_instance == m_sample_count);    // can't split in the middle of a sequence
+    // can't split in the middle of a sequence
+    assert(m_sample_count == 0 || m_instance == m_sample_count || m_cmj_params.m_enabled);
```

Comment:
That assert originally checked that:
- Either we didn't know how many samples would be required (`m_sample_count == 0`)
- Or that we know how many samples would be required, and we would have computed all of them (`m_instance == m_sample_count`).

Now, it seems that the check is entirely bypassed if `m_cmj_params.m_enabled` is enabled. That doesn't seem right: even when CMJ is enabled, it doesn't change the fact that splitting in place can only occur once all samples have been drawn.

Reply:
I agree with you. But in `uniformpixelrenderer`, a large number is given to the context as long as a sample count. Since their are not compatible together, should I set the instance to 0 ?

Reply:
Then what should I do ? Can I create 2 context ?

Reply:
Or why not using base_instance when cmj ?